### PR TITLE
Made Py3.6 main language

### LINKED ISSUE #1525: Make Python 3.6 official development version
Once #1524 is merged we will support py3 as it must pass for changes to be accepted, however we should move to working around 3.6. 

For this, the coverage, minimal and linter builds should be moved to using 3.6 as the default python version

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1525

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by kain88-de** [comment]
I don't like that we have more then one osx test. One osx test would be enough.

**Comment by richardjgowers** [comment]
@kain88-de we should really do them all, but maybe pragmatically we can just do the 3.6 one

**Comment by richardjgowers** [comment]
Or maybe making the osx builds a cron job, so we get a delayed response to them failing

**Comment by kain88-de** [comment]
What should fail with them?

**Comment by tylerjereddy** [comment]
Lots of passes (good), a few strange hiccups, and maybe some minor issues with the new Py version linting in this PR based on initial discussion with @richardjgowers

**Comment by tylerjereddy** [comment]
I think we're seeing the pytest / xdist multicore error `Different tests were collected between gw1 and gw0` consistently here for some rows because @richardjgowers already tried restarting those specific failures. See my [related note](https://github.com/MDAnalysis/mdanalysis/pull/1524#issuecomment-316265602) in a more recent build matrix PR.

File: .travis.yml [hunk 1]
```diff
@@ -43,14 +38,16 @@ env:
 
 matrix:
   fast_finish: true
+
   include:
     - os : linux
       env: NAME='minimal'
-           MEMLEAK='--with-memleak'
+           PYTHON_VERSION=3.6
```
Comment (kain88-de) [inline .travis.yml]: it shouldn't be necessary that you specify the default version all the time.

**Comment by kain88-de** [comment]
We still depend on nose for a lot of tests that try to load it at runtime when we call specific functions.

File: .travis.yml [hunk 2]
```diff
@@ -17,99 +17,69 @@ env:
...
-    - os: linux
-      env: NAME='full-py3.6'
-           PYTHON_VERSION=3.6
-           CONDA_DEPENDENCIES=${CONDA_ALL_DEPENDENCIES}
+    - os: osx
+      env: PYTHON_VERSION=2.7
 
+    - env: NAME='minimal'
+           CONDA_DEPENDENCIES=${CONDA_MIN_DEPENDENCIES}
+           MAIN_CMD="pytest testsuite/MDAnalysisTests/analysis"
```
Comment (kain88-de) [inline .travis.yml]: Also minimal now only runs on the analysis module. This should keep the time at a minimum

**Comment by kain88-de** [comment]
Here are the nose errors that we still have: https://travis-ci.org/MDAnalysis/mdanalysis/jobs/255755498

**Comment by kain88-de** [comment]
Feel free to squash this to one commit during a merge. I've committed a lot of small test changes here.

**Comment by richardjgowers** [comment]
@kain88-de do you mean that we don't test `core` on minimal?

**Comment by kain88-de** [comment]
Yes.

**Comment by kain88-de** [comment]
I can revert that again.

**Comment by richardjgowers** [comment]
@kain88-de yeah I know we're not meant to use weird imports outside of `analysis`, but I'd rather keep the check there just in case we miss something

**Comment by kain88-de** [comment]
No you are right. I anything it should be test all but the analysis tests

**Comment by kain88-de** [comment]
@orbeckst there is a problem installing hole on osx

**Comment by orbeckst** [comment]
> here is a problem installing hole on osx

I have to do some other boring urgent things this afternoon so not much time right now, can you be more specific if you have a moment?

**Comment by kain88-de** [comment]
```
Downloading hole2-NotForProfit-2.2.004-Darwin-i386.tar.gz from https://www.dropbox.com/s/5mzrsyp48i32je4/hole2-NotForProfit-2.2.004-Darwin-i386.tar.gz?dl=1...

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

curl: (77) error setting certificate verify locations:

  CAfile: /etc/ssl/certs/ca-certificates.crt

  CApath: none

[install_hole.sh] ERROR: Failed to download hole2-NotForProfit-2.2.004-Darwin-i386.tar.gz from https://www.dropbox.com/s/5mzrsyp48i32je4/hole2-NotForProfit-2.2.004-Darwin-i386.tar.gz?dl=1 [1]
```

looks like we don't have the correct binary for travis osx

**Comment by orbeckst** [comment]
I think the hole download issue is a problem with [TLS certificates and curl](https://curl.haxx.se/docs/sslcerts.html). We can try
```
curl --insecure  ...
```
or perhaps the workaround with 
```
# fixing curl on travis/anaconda, see http://stackoverflow.com/a/31060428/334357
export CURL_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
```
is not needed anymore and everything "just works" when we remove this line.

**Comment by kain88-de** [comment]
We might need to rethink the python 3.4 CI tests. Conda-forge and anaconda only support the last two stable release branches 3.6 and 3.5. This results for example in numpy 1.13 not being build for python 3.4 as a conda package, and conflicts installing our dependencies.  In turn conflicts during installation are not good because this means we have to wait ages for conda to finish and might cross the 50min mark to single jobs on travis.

File: .travis.yml [hunk 3]
```diff
@@ -28,7 +28,7 @@ env:
     - BUILD_CMD="pip install -v package/ && pip install testsuite/"
     # we need nose for assert_raises and stuff
     - CONDA_MIN_DEPENDENCIES="mmtf-python mock six biopython networkx cython joblib matplotlib scipy griddataformats hypothesis nose"
-    - CONDA_DEPENDENCIES="${CONDA_MIN_DEPENDENCIES} seaborn clustalw=2.1 netcdf4 scikit-learn"
```
Comment (kain88-de) [inline .travis.yml]: Hups coveralls is not equal to coverage. Sorry

**Comment by richardjgowers** [comment]
this is passing, can someone merge this?

**Pull request merged**